roles/tests-integration/Cargo.toml

```diff
@@ -0,0 +1,9 @@
+[package]
+name = "integration-test"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+
+[[bin]]
+name = "run"
```

Comment:
@plebhash try to checkout to this branch and run `cargo build` in `roles` workspace, it should work. but if you remove this bin and the `src` folder, you will see an error. let me know if thats not the behavior you are seeing.

Reply:
@GitGab19 ^^

Reply:
If I remove `[[bin]]` and `src` I get the error @jbesraa was mentioning.
I think the other way is to specify the `common/mod.rs` as a lib there.

I did some tests where I removed `src`, and I put this into `Cargo.toml`:
```
[lib]
path = "tests/common/mod.rs"
```
In this way everything is compiling, and we are explicitly saying that `integration-test` is not a binary

Reply:
Following @GitGab19's suggestion, I was able to successfully build the roles. It seems more efficient to have the integration tests as a lib, as this allows us to remove the redundant `run.rs` file. Interestingly, Fedimint follows a similar workspace structure and also organizes their integration tests as libs, as shown here: https://github.com/fedimint/fedimint/tree/master/fedimint-testing-core.

Reply:
I set my `rustc` to 1.75.0 and I still cannot reproduce this build error

anyways, I agree with the suggestions above

using `lib` feels more coherent than having an empty `run.rs` file

Reply:
Do we need a `bin/` at all? Looking at [ldk-node](https://github.com/lightningdevkit/ldk-node/tree/main/tests), it does not have any executable.

Reply:
we need either a `[lib]` or a `[bin]` statement on `Cargo.toml` of the integration tests crate

the debate here is around the fact that @jbesraa added a placeholder`src/bin/run.rs` so that he could have a `[bin]`, but @GitGab19 suggested to use `[lib]` with a file we already have, which is `tests/common/mod.rs`

to elaborate a bit on what I said on this comment https://github.com/stratum-mining/stratum/pull/1155#issuecomment-2339022184, `ldk-node` will be a good reference for these kind of patterns in a hypothetical future where we have all `roles` unified under one single crate

for now, it is not a good reference, because following its patterns would imply that we would need to unify all crates from `roles` into a single crate